### DISCUSSION TITLE
Edit Existing Templa

### DISCUSSION THREAD

**Discussion by gokhancetiner** [uicrooks/shopify-theme-lab#31] [Q&A]
Hello,

Thanks for the amazing job !!! I really like it. 

May I wonder can we use this template for editing the existing template? or it designed just for a new template from scratch?

Many Thanks,

**Comment by sergejcodes** [comment]
Hi @gokhancetiner  👋🏻. 

It depends on the theme, but it should be possible to implement an existing theme by tweaking some settings in this repo.

I'm assuming you're trying to implement something like the default `Debut` theme from Shopify. If so, I would do the following:

1. [Initialize](https://github.com/uicrooks/shopify-theme-lab#getting-started) the project on remote store
2. Remove [Tailwind CSS](https://github.com/uicrooks/shopify-theme-lab#removing-tailwind-css) and [Vue](https://github.com/uicrooks/shopify-theme-lab#removing-vue)
3. Replace the contents from the `shopify/` directory with the files from the theme in question
4. In [webpack.common.js](https://github.com/uicrooks/shopify-theme-lab/blob/master/.config/webpack/webpack.common.js) replace:
```js
...
new CleanWebpackPlugin({
  cleanOnceBeforeBuildPatterns: ['**/*', '!+(*)static+(*)'],
})
...
```
with

```js
...
new CleanWebpackPlugin({
  cleanOnceBeforeBuildPatterns: ['**/*bundle*'],
})
...
```

5. If you want to have the remote-reloading feature, add the following code before the closing `body` tag:
```liquid
{{ 'bundle.js' | asset_url | script_tag }}
```
6. Run `npm run deploy:dev` to upload all new files
7. Run  `npm run start`

This should be about it.

**Reply by gokhancetiner** [reply]
Hi @sergejcodes,

Thank you for a quick reply, 

May I clarify the question I want to add a new feature to an existing theme like the debut by using TailwindCSS and VueJS? 
Webpack part that you write in my pocket. That's okay. 

Several questions on my mind below. 

1) Where should I put bundle.js and bundle.css in the debut theme? 
2) Should I cover the whole team with VueJS root(I mean put id="App" on theme.liquid)  
3) Should we change VueJS delimiters because liquid and VueJS use the same signs to binding ('{{''}}')

**Reply by sergejcodes** [reply]
1. The structure should be as illustrated below. You want the webpack JavaScript bundle to be the last script before the closing `body` tag. As for Tailwind: probably before the Shopify plugins hook. Also, Tailwind comes out of the box with CSS reset you might want to disable that.

```liquid
<!doctype html>
<html>
  <head>
    ...

    <!-- production css -->
    {{ 'bundle.css' | asset_url | stylesheet_tag }}

    <!-- header hook for Shopify plugins -->
    {{ content_for_header }}
  </head>

  <body>
    <div id="app">
      <!-- everything goes inside here -->
    </div>

    <!-- webpack bundle -->
    {{ 'bundle.js' | asset_url | script_tag }}
  </body>
</html>
```

2. Yes, ideally you want the entire theme to be inside `<div id="app">...</div>`, like in the code example above. Maybe you have to adjust some parts of the Debut theme code to make it work. See [Limitations](https://github.com/uicrooks/shopify-theme-lab#limitations). 
Alternatively, you can instead create new Vue instances for smaller parts of code and select them like in [main.js](https://github.com/uicrooks/shopify-theme-lab/blob/499816d35662db7de8d810c5ba17b097c9cdc82e/src/main.js#L98). That way you don't have to wrap the entire Theme in a Vue instance.

3. You can either change the Vue delimiters or use the Liquid `{% raw %}...{% endraw %}` [tags](https://shopify.github.io/liquid/tags/raw/). Have a look at [vue-examples.liquid](https://github.com/uicrooks/shopify-theme-lab/blob/master/shopify/snippets/vue-examples.liquid), it illustrates a couple of different ways you can use Liquid and Vue together.

**Comment by gokhancetiner** [comment]
Many Thanks, quick and detailed explanation @sergejcodes 

I will play a little with themes with your comments.
